### DISCUSSION TITLE
Controlling temperature

### DISCUSSION THREAD

**Discussion by jlauzon9** [theyosh/TerrariumPI#689] [Q&A] [answered]
Good day,

Amazing software, thank you for developing it.  I am in the process of building and programming (via TerrariumPI) a bioactive enclosure.  I live in a bit of a cold city and want some supplemental heat for cold winters or when my partner blasts the AC too much at night.

I'm struggling to figure out how I can control a ceramic heat emitter with a BME280 temperature probe.  It seems all the relay control I get in the area setup page is based on timers.  Based on the text from the Setup guide:

_"The system uses the sensors alarm values to determine the right conditions you want. For example, using an area with a heater relay and temperature sensors will try to get the temperature in the middle of the minimum and maximum sensor values. Either by using a dimmer or by just toggling on and off. By changing the sensor alarm values, you can raise or lower the wanted temperature. This is the basic logic we use."_

...am I right in assuming if I want to maintain, lets say... 25*c I should set my low alarm at 20 on the BME280, and the high alarm at 30?  And apply a -2offset for nighttime if I want it to be 23*C?   And in addition, I would not need to set a _power on_ or _settle time_ value?

I would set the relay for this heat emitter in the low alarm tab, but not do anything else?  

Thanks to anyone who can confirm or correct me.

**Comment by theyosh** [comment, marked as answer]
Hi, I guess you did not use the area type `heating`? Because that should give you a `sensor` mode.

You are right with the 20 and 30C degrees. The settle time and power on fields needs to be entered `0`. And then it should work like you wanted.

**Reply by theyosh** [reply]
Any updates on this?

**Reply by jlauzon9** [reply]
Got it set up. How long after it reaches temperature should the software shut the relay for the heater off?

I was set at 20-24 low/high alarm, and the ceramic heat emitter was still on with the temp around 23-24 degrees Celsius. I am using a Kasa Smart Dimmer Outlet (but i don't think it is recognized as a Dimmer with TerrariumPi.
